### PR TITLE
6727 i leapp module parser with minimal parsing of plugins

### PR DESCRIPTION
**Pull request by markmckinnon**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+import org.sleuthkit.autopsy.ingest.IngestJobContext;
+import org.sleuthkit.autopsy.ingest.IngestMessage;
+import org.sleuthkit.autopsy.ingest.IngestServices;
+import org.sleuthkit.autopsy.ingest.IngestModule.IngestModuleException;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * Data source ingest module that runs Plaso against the image.
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:54]: I think this is cut and paste error.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+        statusHelper.progress(Bundle.ILeappAnalyzerIngestModule_starting_iLeapp(), 0);
+
+        List<AbstractFile> iLeappFilesToProcess = findiLeappFilesToProcess(dataSource);
+
+        statusHelper.switchToDeterminate(iLeappFilesToProcess.size());
+
+        Integer filesProcessedCount = 0;
+
+        if (!iLeappFilesToProcess.isEmpty()) {
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:121]: Is this check necessary? I think you can just let the for loop recognize there is nothing to process.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+        List<AbstractFile> iLeappFiles = new ArrayList<>();
+
+        FileManager fileManager = getCurrentCase().getServices().getFileManager();
+
+        // findFiles use the SQL wildcard # in the file name
+        try {
+            iLeappFiles = fileManager.findFiles(dataSource, "%", "/"); //NON-NLS
+        } catch (TskCoreException ex) {
+            //Change this
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:192]: I think you can remove this comment, assuming you did change it

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+     */
+    private void addILeappReportToReports(Path iLeappOutputDir, Case currentCase) {
+        List<String> allIndexFiles = new ArrayList<>();
+
+        try (Stream<Path> walk = Files.walk(iLeappOutputDir)) {
+
+            allIndexFiles = walk.map(x -> x.toString())
+                    .filter(f -> f.toLowerCase().endsWith("index.html")).collect(Collectors.toList());
+
+            if (allIndexFiles.get(0).length() > 2) {
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:252]: The first condition should check not empty.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+                currentCase.addReport(allIndexFiles.get(0), MODULE_NAME, Bundle.ILeappAnalyzerIngestModule_report_name());
+            }
+
+        } catch (IOException | TskCoreException ex) {
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:256]: You also need to catch UncheckedIOException, a runtime wrapper for IOException. The IOException caught here, according to the Files.walk documentation, is just related to accessing the iLeappOutputDir. If there is a problem while iterating, then UncheckedIOException will be thrown. Checked exception support was never worked into Java 8 streams and lambdas. You can log the underlying IOException by calling ex.getCause() or something like that.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskException;
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * Data source ingest module that runs Plaso against the image.
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java:64]: Same here as well.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java [hunk 7]
```diff
@@ -0,0 +1,447 @@
...
+        "ILeappFileProcessor.has.run=iLeapp",
+        "ILeappFileProcessor.iLeapp.cancelled=iLeapp run was canceled",
+        "ILeappFileProcessor.completed=iLeapp Processing Completed",
+        "ILeappFileProcessor.error.reading.iLeapp.directory=Error reading iLeapp Output Directory"})
+
+    public ProcessResult processFiles(Content dataSource, Path moduleOutputPath, AbstractFile iLeappFile) {
+
+        try {
+            List<String> iLeappTsvOutputFiles = findTsvFiles(moduleOutputPath);
+            if (!iLeappTsvOutputFiles.isEmpty()) {
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java:103]: This check is not necessary.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java [hunk 8]
```diff
@@ -0,0 +1,447 @@
...
+            allTsvFiles = walk.map(x -> x.toString())
+                    .filter(f -> f.toLowerCase().endsWith(".tsv")).collect(Collectors.toList());
+
+            for (String tsvFile : allTsvFiles) {
+                if (tsvFiles.containsKey(FilenameUtils.getName(tsvFile))) {
+                    foundTsvFiles.add(tsvFile);
+                }
+            }
+
+        } catch (IOException e) {
```
Comment (dannysmyda) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java:133]: See previous comments on Files.walk() in the ingest module, UncheckedIOException should be caught here too.

**Review by dannysmyda — CHANGES_REQUESTED** [review]

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 1 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java]: Yes it is.  Fixed

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 2 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java]: It was removed and ran fine

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 3 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java]: Comment removed

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 4 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java]: Now checks !allIndexFiles.isEmpty()

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 5 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java]: Added exception

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java (hunk 7 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java]: Removed check

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java (hunk 6 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java]: Changed comment

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java (hunk 8 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappFileProcessor.java]: Added check for exception

**Review by dannysmyda — APPROVED** [review]

**Pull request merged**